src/gravel/controllers/nodes/deployment.py

```diff
@@ -265,3 +288,79 @@ async def join(
 
         self._state.mark_ready()
         return True
+
+    async def bootstrap(
+        self,
+        config: DeploymentBootstrapConfig,
+        finisher: Callable[[bool, Optional[str]], Awaitable[None]]
+    ) -> None:
+
+        assert config.hostname
+        assert config.address
+        assert config.token
+
+        hostname = config.hostname
+        address = config.address
+        token = config.token
+
+        if self._state.error:
+            raise NodeCantBootstrapError("node is in error state")
+
+        try:
+            await self._prepare_bootstrap(hostname, address, token)
+        except NodeError as e:
+            logger.error(f"bootstrap prepare error: {e.message}")
+            raise e
+
+        assert not self._bootstrapper
+        await self._start_bootstrap()
+        self._bootstrapper = Bootstrap()
+
+        async def finish_bootstrap_cb(
+            success: bool,
+            error: Optional[str]
+        ) -> None:
+            if not success:
+                logger.error(f"bootstrap finish error: {error}")
+                assert self._state.bootstrapping
+                self._state.mark_error()
+
+            await finisher(success, error)
+
+        try:
+            await self._bootstrapper.bootstrap(
+                address, finish_bootstrap_cb
+            )
+        except BootstrapError as e:
+            logger.error(f"bootstrap error: {e.message}")
+            raise NodeCantBootstrapError(e.message)
+
+    async def _prepare_bootstrap(
+        self,
+        hostname: str,
+        address: str,
+        token: str
+    ) -> None:
+        assert self._state
+        if self._state.bootstrapping:
+            raise NodeCantBootstrapError("node bootstrapping")
+        elif not self._state.nostage:
+            raise NodeCantBootstrapError("node can't be bootstrapped")
+
+        await spawn_etcd(
+            self._gstate,
+            new=True,
+            token=token,
+            hostname=hostname,
+            address=address
+        )
+
+    async def _start_bootstrap(self) -> None:
```

Comment:
(nit) Similarly I find this confusing. I know it's denoted as a private method, but if you ignore that it's confusing if I should "bootstrap" the node or "start the bootstrap".. This could probably just be folded into the bootstrap method rather than being its own method.